src/rate-limiters/redis.js

```diff
@@ -0,0 +1,102 @@
+/* @flow */
+/* global Bluebird$Promise Logger RedisClient*/
+'use strict'
+const joi = require('joi')
+const RateLimiter = require('ratelimiter')
+const Promise = require('bluebird')
+const redis = require('redis')
+
+const optsSchema = joi.object({
+  durationMs: joi.number().integer().min(0).required(),
+  host: joi.string().required(),
+  log: joi.object().required(),
+  port: joi.string().required()
+})
+
+module.exports = class RedisRateLimiter {
+  port: string;
+  host: string;
+  log: Logger;
+  durationMs: number;
+  client: RedisClient;
+
+  /**
+   * creates RedisRateLimiter object
+   * @param  {Object} opts  redis connection options
+   * @param  {String} opts.port redis connection port
+   * @param  {String} opts.host redis connection host
+   * @param  {Logger} opts.log  worker server logger
+   * @return {RedisRateLimiter}
+   */
+  constructor (opts: Object) {
+    this.port = opts.port ||
+      process.env.REDIS_PORT ||
+      '6379'
+
+    this.host = opts.host ||
+      process.env.REDIS_HOST ||
+      'localhost'
+
+    this.durationMs = opts.durationMs ||
+      parseInt(process.env.RATE_LIMIT_DURATION, 10) ||
+      1000
+
+    this.log = opts.log
+
+    joi.assert(this, optsSchema)
+  }
+
+  /**
+   * Connect redis client to redis
+   * @return {Promise}
+   * @resolves {undefined} When connection is ready
+   * @reject {Error} When there was an error connecting
+   */
+  connect (): Bluebird$Promise<*> {
+    return Promise.fromCallback((cb) => {
+      this.log.trace('connecting to redis')
+      this.client = redis.createClient(this.port, this.host)
+      this.client.on('error', cb)
```

Comment:
some potential problem here that should be checked.
I'm not sure that redis client would emit either `ready` or `error` (this should be checked in docs). I think it can emit `ready` and later `error`. If that is the case I think Promise wouldn't be rejected.